l1-contracts/src/core/libraries/Decoder.sol

```diff
@@ -511,7 +510,12 @@ library Decoder {
       kernelPublicInputsLogsHash = sha256(abi.encodePacked(logsHashes));
     }
 
-    return (kernelPublicInputsLogsHash, offset);
+    uint256 offsetInBlock;
+    assembly {
+      offsetInBlock := sub(offset, _l2Block.offset)
+    }
```

Comment:
Do we need assembly for this?

Reply:
Since requesting offset of a value in calldata is something you don't ever need if you are not using assembly it's not available directly in Solidity.